### ISSUE TITLE
Implement new Concept Exercise: generic-types

### ISSUE THREAD

**Issue by workingjubilee** [exercism/rust#1071]
This issue describes how to implement the `generic-types` concept exercise for the rust track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

Often called simply "generics", or more explicitly _generic type parameters_, generics are important in writing clear and DRY code. The student has likely used a generic container by this point to solve one of the other exercises, and so should be introduced to them formally.

## Learning objectives

- recognize the syntax for generics
- recognize the convention of using T for a type parameter

## Out of scope

- minimize discussion of generics outside functions and structs
- so avoid generic lifetimes
- or generic traits

## Concepts

- generics
- generic types

## Prerequisites

This will require a conceptual understanding of
- complex types
- parameters
- type annotations

### Hints

- [Generic Types]

### After

- [Generic Types]

## Representer

No changes required

## Analyzer

No changes required

## Implementing

See our [implementation guide](https://github.com/exercism/v3/blob/master/languages/rust/howto/implementing-a-concept-exercise.md#how-to-implement-a-rust-concept-exercise) for guidance.  In addition you may want to make use of the shell script while in the `concept` directory, with the argument of the exercise name, e.g. `sh boil_exercise.sh generic-types`

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

[Generic Types]: https://doc.rust-lang.org/book/ch10-00-generics.html

**Comment by senekor** [comment]
After several failed attempts to create a high-quality syllabus, I believe any future attempts will have to do their own design work from scratch.
